flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -109,27 +114,51 @@
 
 	private HiveMetastoreClientWrapper client;
 
-	public HiveCatalog(String catalogName, String hivemetastoreURI) {
-		this(catalogName, DEFAULT_DB, getHiveConf(hivemetastoreURI));
+	public HiveCatalog(String catalogName, @Nullable String defaultDatabase, @Nullable String hiveSiteFilePath) {
+		this(catalogName,
+			defaultDatabase == null ? DEFAULT_DB : defaultDatabase,
+			getHiveConf(hiveSiteFilePath));
 	}
 
-	public HiveCatalog(String catalogName, HiveConf hiveConf) {
-		this(catalogName, DEFAULT_DB, hiveConf);
+	public HiveCatalog(String catalogName, @Nullable String defaultDatabase, @Nullable URL hiveSiteUrl) {
+		this(catalogName,
+			defaultDatabase == null ? DEFAULT_DB : defaultDatabase,
+			getHiveConf(hiveSiteUrl));
 	}
 
-	public HiveCatalog(String catalogName, String defaultDatabase, HiveConf hiveConf) {
-		super(catalogName, defaultDatabase);
-		this.hiveConf = checkNotNull(hiveConf, "hiveConf cannot be null");
+	public HiveCatalog(String catalogName, @Nullable String defaultDatabase, @Nullable HiveConf hiveConf) {
+		super(catalogName, defaultDatabase == null ? DEFAULT_DB : defaultDatabase);
+
+		this.hiveConf = hiveConf == null ? getHiveConf("") : hiveConf;
 
 		LOG.info("Created HiveCatalog '{}'", catalogName);
 	}
 
-	private static HiveConf getHiveConf(String hiveMetastoreURI) {
-		checkArgument(!StringUtils.isNullOrWhitespaceOnly(hiveMetastoreURI), "hiveMetastoreURI cannot be null or empty");
+	private static HiveConf getHiveConf(String filePath) {
+
+		URL url = null;
+
+		if (!StringUtils.isNullOrWhitespaceOnly(filePath)) {
+			try {
+				url = new File(filePath).toURI().toURL();
+
+				LOG.info("Successfully loaded '{}'", filePath);
+
+			} catch (MalformedURLException e) {
+				throw new CatalogException(
+					String.format("Failed to get hive-site.xml from the given path '%s'", filePath), e);
+			}
+		}
+
+		return getHiveConf(url);
+	}
+
+	private static HiveConf getHiveConf(URL hiveSiteUrl) {
+		LOG.info("Setting hive-site location as {}", hiveSiteUrl);
+
+		HiveConf.setHiveSiteLocation(hiveSiteUrl);
```

Comment:
Nit: it's probably better if we don't rely Hive to handle null. We can call HiveConf.setHiveSiteLocation() only if hiveSiteUrl is not null.

Reply:
Since hive site url is a static member variable, we need to set it even when it's null, in order to support multi hive catalogs. Say 1st catalog has hive site and 2nd doesn't. Not setting it as null for the 2nd would make the 2nd reuse the 1st's hive site file.

Reply:
Wondering will this incur any concurrency issue, e.g. when user access tables from different Hive catalogs? I think at least some constructors of HiveConf is no longer safe to use after this change, like `HiveConf()` and `HiveConf(Class<?> cls)` which automatically load the specified hive-site.xml.
I think it's better not to rely on static fields of HiveConf. Instead, we can load the hive-site (perhaps `Properties::loadFromXML`?) ourselves and `HiveConf::hiveSiteURL` should always be set to null.

Reply:
maybe we should reset the HiveConf's hiveSiteUrl back to `null` after creating the HiveConf instance?

Reply:
That's also a viable solution, in which case we need a unified place to create HiveConf instances.